ff/src/biginteger/mod.rs

```diff
@@ -137,4 +138,18 @@ pub trait BigInteger:
         *self = Self::read(reader)?;
         Ok(())
     }
+
+    /// Takes two slices of u64 representing big integers and returns a bigger
+    /// BigInteger of type Self representing their product. Preferably used
+    /// only for even NUM_LIMBS. We require the invariant that this.len() ==
+    /// other.len() == NUM_LIMBS / 2
+    fn mul_no_reduce(this: &[u64], other: &[u64]) -> Self;
```

Comment:
Does this need to be a method on bigint? Can't it just be a free-standing function that is generic over `B: BigInteger`?

So something like `fn mul_no_reduce<B: BigInteger>(this: &[u64], other: &[u64]) -> B`?

Reply:
Hmm definitely this is possible. But this way, one doesn't need to import additional methods

Reply:
Is your concern that there would be too many methods?

Reply:
Yeah, I think I want to reduce the API surface of our existing traits, because implementing them is kind of a pain right now.

Reply:
How do you use this API in your later PRs? Maybe it's possible to come up with an API that fits slightly better into the rest of the methods, as the current design API works with slices and only for for even-limb BigInts.

Reply:
This (impl pain) isn't really an issue for the BigInteger trait, however? 

The question is whether one sees potential additional use elsewhere.

But, as the impact is small, I can move it to a separate method now, under an empty struct `BigIntegerHelper`